tools/generator/src/DTO/XcodeScheme.swift

```diff
@@ -114,6 +118,33 @@ disabled, but the target is referenced in the scheme's \(keyPath.actionType) act
     }
 }
 
+// MARK: PrePostAction
+
+extension XcodeScheme {
+    struct PrePostAction: Equatable, Decodable {
+        let name: String
+        let expandVariablesBasedOn: VariableExpansionContext?
```

Comment:
This should not be optional. The `VariableExpansionContext` enum has a `none` value.

Reply:
Hey! I made this optional because the settings provider(or target) can be passed as None. This is also consistent with the type that the `TestAction` has for the same reason. 

In case we want to make it a non-optional, the bazel side of it has to convert None to "none" string OR we would have to come up with a custom decodable that handles the case when Bazel passes None(serialized to "null") but we expect it to be a non-optional.

I am more inclined in keeping it as an optional above (`VariableExpansionContext`) as the intent is pretty clear when None is passed (without we converting None to "none" and other custom decodable workarounds). PrePostActionInfo will have a non-optional `VariableExpansionContextInfo` though.

Let me know your thoughts and I can change the implementation accordingly.

Reply:
Gotcha. I thought that we provided a value in Starlark, if `None` was specified. Changing that behavior is definitely outside the scope of your PR. Thanks.